### PR TITLE
Cover all use cases at caBundle

### PR DESCRIPTION
**Pull request by qinqon**
The code that reconcile CABundle at webhookconfiguration was covering only the
case when webhookcofiguration is updated but not when CABundle is updated by a
rotation or created after a webhook delete.

Signed-off-by: Quique Llorente <ellorent@redhat.com>

### CODE REVIEW

File: pkg/apply/merge_test.go [hunk 1]
```diff
@@ -374,20 +337,131 @@ webhooks:
...
+			err := apply.MergeMetadataForUpdate(current, updated)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updated.GetCreationTimestamp()).To(Equal(current.GetCreationTimestamp()))
+			Expect(updated.GetGeneration()).To(Equal(current.GetGeneration()))
+			Expect(updated.GetResourceVersion()).To(Equal(current.GetResourceVersion()))
+			Expect(updated.GetSelfLink()).To(Equal(current.GetSelfLink()))
+			Expect(updated.GetUID()).To(Equal(current.GetUID()))
 		})
 	})
+
```
Comment (RamLavi) [inline pkg/apply/merge_test.go:466]: please drop the extra line

File: pkg/apply/merge.go [hunk 2]
```diff
@@ -178,20 +178,23 @@ func mergeWebhookConfiguration(current, updated *unstructured.Unstructured) erro
...
 		}
 
-		err = unstructured.SetNestedField(updatedWebhook.(map[string]interface{}), caBundle, "clientConfig", "caBundle")
+		_, updatedCABundleFound, err := unstructured.NestedString(updatedWebhook.(map[string]interface{}), "clientConfig", "caBundle")
 		if err != nil {
-			return errors.Wrapf(err, "failed copying caBundle from current config to updated config at webhook %s", updatedWebhookName)
+			return errors.Wrapf(err, "failed searching updated caBundle 'field' at webhook %s", updatedWebhookName)
 		}
 
+		// If CABundle was already set and updated config does not have it's keept
```
Comment (phoracek) [inline pkg/apply/merge.go:191]: I don't understand this comment

File: pkg/apply/merge.go [hunk 3]
```diff
@@ -178,20 +178,23 @@ func mergeWebhookConfiguration(current, updated *unstructured.Unstructured) erro
 			continue
 		}
 
-		// If there is caBundle set in the current config, we want to keep that value
```
Comment (phoracek) [inline pkg/apply/merge.go:181]: Where did this comment go, don't we still need it?

File: pkg/apply/merge.go [hunk 4]
```diff
@@ -178,20 +178,23 @@ func mergeWebhookConfiguration(current, updated *unstructured.Unstructured) erro
...
-		caBundle, found, err := unstructured.NestedString(currentWebhook, "clientConfig", "caBundle")
+		currentCABundle, currentCABundleFound, err := unstructured.NestedString(currentWebhook, "clientConfig", "caBundle")
```
Comment (phoracek) [inline pkg/apply/merge.go:181]: The commit message suggests this patch is only testing stuff. But you are also fixing bugs in the code. Please express it in the commit message. Say what was wrong and how is it fixed.

**Pull request merged**